src/ctim/domain/observables/ip.clj

```diff
@@ -54,18 +53,38 @@
    "2620:4f:8000::/48"
    "fe80::/10"])
 
+(s/defn valid-ipv4? :- s/Bool
+  "Is this a valid ipv4 address?"
+  [ip-str :- s/Str]
+  (and (string? ip-str)
```

Comment:
I think it would be nicer to log when those function are wrongly used.

```suggestion
  (and (or (string? ip-str) (log/error "valid-ipv4? called with a non string value"))
```